ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -0,0 +1,822 @@
+      numGenKeys :: Word8
+      numGenKeys =
+        case length (Genesis.unGenesisWStakeholders $ Genesis.configBootStakeholders cfg) of
+          n
+            | n > fromIntegral (maxBound :: Word8) -> panic
+              "updateBody: Too many genesis keys"
+            | otherwise -> fromIntegral n
+
+      delegationMap =
+        V.Interface.delegationMap
+        $ CC.Block.cvsDelegationState state
+
+      fixPMI pmi = reAnnotate $ Annotated pmi ()
+
+  ledgerTipPoint (ByronLedgerState state _) =
+      Point { pointSlot = convertSlot (CC.Block.cvsLastSlot state)
+            , pointHash = case CC.Block.cvsPreviousHash state of
+                            Left _genHash -> GenesisHash
+                            Right hdrHash -> BlockHash hdrHash
+            }
+
+{-------------------------------------------------------------------------------
+  Support for PBFT consensus algorithm
+-------------------------------------------------------------------------------}
+
+type instance BlockProtocol (ByronBlock cfg) = ExtNodeConfig cfg (PBft PBftCardanoCrypto)
+
+type instance BlockProtocol (ByronHeader cfg) = ExtNodeConfig cfg (PBft PBftCardanoCrypto)
+
+instance (Given CC.Slot.EpochSlots, Given CC.Block.HeaderHash, Typeable cfg) => HasPreHeader (ByronBlock cfg) where
+  type PreHeader (ByronBlock cfg) = CC.Block.ToSign
+  blockPreHeader = unAnnotated . CC.Block.recoverSignedBytes given
+                   . CC.Block.blockHeader . unByronBlock
+
+-- TODO get rid of this once we have a BlockHeader type family
+instance (Given CC.Slot.EpochSlots, Given CC.Block.HeaderHash, Typeable cfg) => HasPreHeader (ByronHeader cfg) where
+  type PreHeader (ByronHeader cfg) = CC.Block.ToSign
+  blockPreHeader = unAnnotated . CC.Block.recoverSignedBytes given
+                   . unByronHeader
+
+-- TODO get rid of this once we have a BlockHeader type family
+instance (Given CC.Slot.EpochSlots, Given CC.Block.HeaderHash, Typeable cfg) => HasPayload (PBft PBftCardanoCrypto) (ByronHeader cfg) where
+  blockPayload _ (ByronHeader header) = PBftPayload
+    { pbftIssuer = VerKeyCardanoDSIGN
+                   . Crypto.pskDelegateVK
+                   . Crypto.psigPsk
+                   . CC.Block.unBlockSignature
+                   . CC.Block.headerSignature
+                   $ header
+    , pbftSignature = SignedDSIGN
+                      . SigCardanoDSIGN
+                      . Crypto.Signature
+                      . Crypto.psigSig
+                      . CC.Block.unBlockSignature
+                      . CC.Block.headerSignature
+                      $ header
+    }
+
+
+instance (Given CC.Slot.EpochSlots, Given CC.Block.HeaderHash, Typeable cfg) => HasPayload (PBft PBftCardanoCrypto) (ByronBlock cfg) where
+  blockPayload cfg = blockPayload cfg . byronHeader
+
+-- | Override the delegation map from the ledger view
+--
+-- This is to work around a bug in cardano-ledger
+-- <https://github.com/input-output-hk/cardano-ledger/issues/504>
```

Comment:
This function is not called anymore, I think. We should give this module an explicit export list.